fakeldap.py

```diff
@@ -29,7 +29,15 @@
 import logging
 import types
 from collections import defaultdict
-import ldap
+
+try:
+    import ldap
+except ImportError:
+    class ldap(object):
+        class LDAPError(Exception): pass
+        class INVALID_CREDENTIALS(LDAPError): pass
+        class NO_SUCH_OBJECT(LDAPError): pass
+        class ALREADY_EXISTS(LDAPError): pass
```

Comment:
Can you add a comment here explaining the motivation from the PR description?  I think it'll be helpful for future folks reading the code to not need to spelink.